framework/base/ErrorException.php

```diff
@@ -44,7 +44,8 @@ public function __construct($message = '', $code = 0, $severity = 1, $filename =
 
         if (function_exists('xdebug_get_function_stack')) {
             $trace = array_slice(array_reverse(xdebug_get_function_stack()), 3, -1);
-            foreach ($trace as &$frame) {
+            $phpCompatibleTrace = [];
```

Comment:
The code is quite cryptic w/o a comment pointing to the issue. It's likely that it will be "optimized" in the future so I'd add a comment.